client/account/account.go

```diff
@@ -0,0 +1,66 @@
+package account
+
+import (
+	"fmt"
+
+	"github.com/decred/dcrd/dcrec/secp256k1"
+	"github.com/decred/dcrdex/server/account"
+	"github.com/decred/dcrdex/server/account/pki"
+)
+
+// Account represents a dex account.
+type Account struct {
+	ID      account.AccountID
+	PrivKey *secp256k1.PrivateKey
+	PubKey  *secp256k1.PublicKey
+}
+
+// New creates a dex account. The associated private key of the
+// account is generated randomly.
+func New() (*Account, error) {
+	privKey, err := secp256k1.GeneratePrivateKey()
+	if err != nil {
+		return nil, err
+	}
+
+	pubKey := privKey.PubKey()
+
+	var pkb [pki.PubKeySize]byte
+	copy(pkb[:], pubKey.SerializeCompressed()[:])
+
+	return &Account{
+		ID:      account.NewID(pkb),
+		PrivKey: privKey,
+		PubKey:  pubKey,
+	}, nil
+}
+
+// NewAccountFromPrivKey creates a dex account from the provided private
+// key bytes.
+func NewAccountFromPrivKey(privK [pki.PrivKeySize]byte) (*Account, error) {
+	privKey, pubKey := secp256k1.PrivKeyFromBytes(privK[:])
+
+	var pkb [pki.PubKeySize]byte
+	copy(pkb[:], pubKey.SerializeCompressed()[:])
+
+	return &Account{
+		ID:      account.NewID(pkb),
```

Comment:
If `NewID` also is changed to take a slice instead of an array, the above `pkb` copy isn't needed either.